components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/notification/DefaultNotificationChannelManager.java

```diff
@@ -0,0 +1,351 @@
+
+        // If there are no values for channel related claims.
+        if (MapUtils.isEmpty(claimMap)) {
+            if (log.isDebugEnabled()) {
+                String error = String.format("No notification channel for user : %1$s with domain : %2$s.",
+                        userstoreDomain + CarbonConstants.DOMAIN_SEPARATOR + username, tenantDomain);
+                log.debug(error);
+            }
+            throw new NotificationChannelManagerClientException(
+                    IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_NOTIFICATION_CHANNELS.getCode(),
+                    IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_NOTIFICATION_CHANNELS.getMessage());
+        }
+        String preferredChannel = claimMap.get(IdentityMgtConstants.Claim.PREFERED_CHANNEL_CLAIM);
+
+        // If the user has already specified a channel, then no need to evaluate the claims for a notification channel.
+        if (StringUtils.isNotEmpty(preferredChannel)) {
+            // Check whether the preferred channel has claims values.
+            if (validatePreferredChannelWithValuesInClaimMap(preferredChannel, claimMap)) {
+                return preferredChannel;
+            }
+            throw new NotificationChannelManagerClientException(
+                    IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_CLAIM_MATCHED_FOR_PREFERRED_CHANNEL.getCode(),
+                    IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_CLAIM_MATCHED_FOR_PREFERRED_CHANNEL.getMessage());
+        }
+        return resolveCommunicationChannel(this.channels, claimMap);
+    }
+
+    /**
+     * Checks whether the preferred communication channel matches the given channel values.
+     *
+     * @param claimsMap User's claims
+     * @return True of preferred channel has a value in the claims map.
+     */
+    private boolean validatePreferredChannelWithValuesInClaimMap(String preferredChannel, Map<String, String> claimsMap)
+            throws NotificationChannelManagerClientException {
+
+        // Get the notification channel which matches the given channel type
+        NotificationChannels channel = NotificationChannels.getNotificationChannel(preferredChannel);
+        return StringUtils.isNotEmpty(claimsMap.get(channel.getClaimUri()));
+    }
+
+    /**
+     * Get a notification channel for the given set of claims which are belong to a user.
+     *
+     * @param channels  notification channels
+     * @param claimsMap Claims
+     * @return notification channel
+     */
+    private String resolveCommunicationChannel(ArrayList<NotificationChannels> channels,
+            Map<String, String> claimsMap) {
+
+        ArrayList<String> availableChannels = new ArrayList<>();
+        for (NotificationChannels channel : channels) {
+
+            // Checks whether there is a value value for given communication channel url.
+            String value = claimsMap.get(channel.getClaimUri());
+            if (StringUtils.isNotEmpty(value)) {
+
```

Comment:
Can remove this empty line.

Reply:
Fixed by https://github.com/wso2-extensions/identity-governance/pull/342